### PR TITLE
TIG-3479 Initial codebase for Ansible configuration

### PR DESCRIPTION
**Pull request by JasurbekNURBOYEV**
This is more like a high level variant of whole codebase. There might be issues as I'm not super cool with Ansible. Added a README.md to help with explanation of what s going on here.

### CODE REVIEW

**Comment by JasurbekNURBOYEV** [comment]
> * Did you get a chance to try integrating a task from Ansible Galaxy Zakhar talked about the other day?
> * Would you mind providing an example of a custom role? Maybe just call it `example_custom` or something?
> * 💅  could the top-level tasks directory be renamed to something like `bootstrap-tasks` to indicate that they're not user-facing ones like git/tasks.

1. I didn't try integrating Ansible Galaxy, our roles tend to exist inside our repo, so I thought there was no need for Ansible Galaxy. I don't really know what other teams will be doing inside roles, so I don't know what roles to take from Ansible Galaxy.
2. The current "dag" and "sap" were supposed to mimic the sample, however they are too simple. I think whoever wants to implement role for their team, they'll need to go through documentation of how roles are defined. Roles can be way too simple (like current "sap" or "dag") if there is no much for team to do while configurations, or it might be too complicated (which I probably can't make an example for). I'll try implementing a team role for Query team, maybe that will serve as a good and practical sample.
3. yeah, I will rename it.

**Comment by guoyr** [comment]
> I didn't try integrating Ansible Galaxy, our roles tend to exist inside our repo, so I thought there was no need for Ansible Galaxy. I don't really know what other teams will be doing inside roles, so I don't know what roles to take from Ansible Galaxy.

Got it. I think we had looked at installing zsh? Could that be a suitable candidate? If there are off the shelf components, we'd want people to use them. I didn't follow all of  the discussion around Galaxy so let me know if there are known limitations

**Comment by JasurbekNURBOYEV** [comment]
> > I didn't try integrating Ansible Galaxy, our roles tend to exist inside our repo, so I thought there was no need for Ansible Galaxy. I don't really know what other teams will be doing inside roles, so I don't know what roles to take from Ansible Galaxy.
> 
> Got it. I think we had looked at installing zsh? Could that be a suitable candidate? If there are off the shelf components, we'd want people to use them. I didn't follow all of the discussion around Galaxy so let me know if there are known limitations
Yeah, we can use `oh-my-zsh` role from Ansible Galaxy. I'll look into that.

File: ansible/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+* tasks /
+        - * task1.yml
+        - * task2.yml
+        - * ...
+vars / main.yml  # configuration. can be modified to tweak the flow.
+* start.yml  # main entry point for ansible      
+```
+
+### Rules to make everything better
+- Use lowercase when naming your team role, `"_"` instead of space. `"My awesome team"` might be good in general,
```
Comment (guoyr) [inline ansible/README.md:33]: 👍

File: ansible/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+### Rules to make everything better
+- Use lowercase when naming your team role, `"_"` instead of space. `"My awesome team"` might be good in general, 
+but we prefer `"my_awesome_team"`
+- Most of the time `roles` folder is what you want to support your team and anything outside this folder is 
+not specific to teams. Unless you are a maintainer or have a duty to fix bugs (not related to specific teams) here, 
+please do not edit any folder or file outside `roles` folder.
+- Use Ansible commands instead of shell commands when possible. 
+Most of the time they are idempotent, causing fewer problems.
+
+### How to Run
```
[the remark this answers is not in the document]
Comment (JasurbekNURBOYEV) [reply ansible/README.md]: ✅

File: ansible/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# Per team virtual workstation configuration
+## Why?
+Teams sometimes (or usually) need to follow instructions to set up their virtual workstations. 
+And some teams (maybe all of them) have manuals to follow. 
+This tiny project tries to automate that process and take away all setup hurdle from users.
+
+## When you add your team to `roles/teams` or edit existing team role
+Please, update `roles/common` folder: add any new common task that can be shared among other teams or can be used 
+by other users (as a role). This folder keeps all roles of all teams. When users can't find their team,
```
Comment (guoyr) [inline ansible/README.md:9]: Rephrase: When users.... -> If a user can't find their team, please create a new one on top of the common roles shared by all teams.

**Review by guoyr — APPROVED** [review]
LGTM!

**Pull request merged**